plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerContainerNameGenerator.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import org.eclipse.che.api.core.model.machine.Machine;
+import org.eclipse.che.api.core.model.machine.MachineConfig;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.commons.user.User;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is used for generation docker container name or parsing important information from docker container name.
+ *
+ * @author Alexander Andrienko
+ */
+public class DockerContainerNameGenerator {
+    private static final String WORKSPACE_ID_GROUP = "(?<workspaceId>workspace[0-9a-z]{16})";
+    private static final String MACHINE_ID_GROUP   = "(?<machineId>machine[0-9a-z]{16})";
+
+    private static final String  CONTAINER_NAME_REGEX   = "^" + WORKSPACE_ID_GROUP + "_" + MACHINE_ID_GROUP + "(_.*){2}$";
```

Comment:
I think it doesn't matter which chars are in container name after valuable for you parts. So you can simplify regexp.